File: app/src/main/java.backup/com/lumiyaviewer/lumiya/react/Subscribable.java
package com.lumiyaviewer.lumiya.react;

import com.lumiyaviewer.lumiya.react.Subscription.OnData;
import com.lumiyaviewer.lumiya.react.Subscription.OnError;
import java.util.concurrent.Executor;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public interface Subscribable<K, T> {
    Subscription<K, T> subscribe(@Nonnull K k, @Nonnull OnData<T> onData);

    Subscription<K, T> subscribe(@Nonnull K k, @Nonnull OnData<T> onData, @Nullable OnError onError);

    Subscription<K, T> subscribe(@Nonnull K k, @Nullable Executor executor, @Nonnull OnData<T> onData);

    Subscription<K, T> subscribe(@Nonnull K k, @Nullable Executor executor, @Nonnull OnData<T> onData, @Nullable OnError onError);
}

File: app/src/main/kotlin/com/lumiyaviewer/lumiya/react/Subscribable.kt
package com.lumiyaviewer.lumiya.react

import com.lumiyaviewer.lumiya.react.Subscription.OnData
import com.lumiyaviewer.lumiya.react.Subscription.OnError
import java.util.concurrent.Executor

interface Subscribable<K, T> {
    fun subscribe(key: K, onData: OnData<T>): Subscription<K, T>
    
    fun subscribe(key: K, onData: OnData<T>, onError: OnError?): Subscription<K, T>
    
    fun subscribe(key: K, executor: Executor?, onData: OnData<T>): Subscription<K, T>
    
    fun subscribe(key: K, executor: Executor?, onData: OnData<T>, onError: OnError?): Subscription<K, T>
}
